contrib/completion/zsh/_docker

```diff
@@ -2316,14 +2316,28 @@ __docker_caching_policy() {
 
 __docker_commands() {
     local cache_policy
+    integer force_invalidation=0
 
     zstyle -s ":completion:${curcontext}:" cache-policy cache_policy
     if [[ -z "$cache_policy" ]]; then
         zstyle ":completion:${curcontext}:" cache-policy __docker_caching_policy
     fi
 
-    if ( [[ ${+_docker_subcommands} -eq 0 ]] || _cache_invalid docker_subcommands) \
-        && ! _retrieve_cache docker_subcommands;
+    if ( [[ ${+_docker_hide_legacy_commands} -eq 0 ]] || _cache_invalid docker_hide_legacy_commands ) \
```

Comment:
Yes. `${+var}` expands to 0 if the variable is not defined and 1 otherwise. Having said that, I see there is some logic errors since you are assigning a value to the variable from another variable. Whatever the content is, with `${+var}`, what's matter is the fact the variable is defined, even if it's content is 0. Maybe it's better to use `(( ${_docker_hide_legacy_commands} ))`. It will be false if the variable is not defined or defined to be 0 or empty. Safer and with the ability to negate is `(( ${_docker_hide_legacy_commands:-0} ))` (use value 0 if the variable is not defined or empty).


Note that I made a typo. This is `(( ! ${+_docker_hide_legacy_commands} ))`